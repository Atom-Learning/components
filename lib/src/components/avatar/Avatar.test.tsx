import { fireEvent, render } from '@testing-library/react'
import { BatteryMedium } from '@atom-learning/icons'
import { axe } from 'jest-axe'
import * as React from 'react'

import { Avatar } from '.'

describe('Avatar component', () => {
  it('renders an image avatar', async () => {
    const { container } = render(
      <Avatar name="a name">
        <Avatar.Image alt="somobody's avatar" src="path-to-image" />
      </Avatar>
    )

    expect(container).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    let view = render(
      <Avatar name="a name">
        <Avatar.Image alt="somobody's avatar" src="path-to-image" />
      </Avatar>
    )
    expect(await axe(view.container)).toHaveNoViolations()

    view = render(
      <Avatar name="a name">
        <Avatar.Initial />
      </Avatar>
    )
    expect(await axe(view.container)).toHaveNoViolations()

    view = render(
      <Avatar name="a name">
        <Avatar.Placeholder />
      </Avatar>
    )
    expect(await axe(view.container)).toHaveNoViolations()

    view = render(
      <Avatar name="a name">
        <Avatar.Icon is={BatteryMedium} />
      </Avatar>
    )
    expect(await axe(view.container)).toHaveNoViolations()
  })

  it('renders the initial of the name when the src of the image is not available and a name was provided', async () => {
    const { container } = render(
      <Avatar name="Bob">
        <Avatar.Image alt="Bob's avatar" src="" />
      </Avatar>
    )

    expect(container).toMatchSnapshot()
  })

  it('renders an xs size with the fallback icon when src and name are missing', async () => {
    const { container } = render(
      <Avatar size="xs">
        <Avatar.Image alt="an avatar" src="" />
      </Avatar>
    )

    expect(container).toMatchSnapshot()
  })

  it('renders an inital avatar', async () => {
    const { container } = render(
      <Avatar size="xs" name="Alice">
        <Avatar.Initial />
      </Avatar>
    )

    expect(container).toMatchSnapshot()
  })

  it('renders a fallback icon instead of an initial if name is missing', () => {
    const { container } = render(
      <Avatar name="">
        <Avatar.Initial />
      </Avatar>
    )

    expect(container).toMatchSnapshot()
  })

  it('renders an icon avatar', () => {
    const { container } = render(
      <Avatar>
        <Avatar.Icon is={BatteryMedium} />
      </Avatar>
    )

    expect(container).toMatchSnapshot()
  })

  it('calls the onClick function passed when clicked', () => {
    const onClick = jest.fn()
    const { container } = render(
      <Avatar onClick={onClick}>
        <Avatar.Icon is={BatteryMedium} />
      </Avatar>
    )
    fireEvent.click(container.firstChild as ChildNode)

    expect(onClick).toHaveBeenCalled()
  })

  it('does not call the onClick function passed when click if disabled', () => {
    const onClick = jest.fn()
    const { container } = render(
      <Avatar disabled onClick={onClick}>
        <Avatar.Icon is={BatteryMedium} />
      </Avatar>
    )
    fireEvent.click(container.firstChild as ChildNode)

    expect(onClick).not.toHaveBeenCalled()
  })
})
