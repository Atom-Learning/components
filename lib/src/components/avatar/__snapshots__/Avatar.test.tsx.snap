// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component renders a fallback icon instead of an initial if name is missing 1`] = `
@media  {
  .c-hMBZYr {
    display: flex;
    justify-content: center;
    align-items: center;
    color: var(--colors-tonal400);
    border-radius: var(--radii-round);
    border: 2px solid var(--colors-tonal100);
    background-color: var(--colors-white);
    overflow: hidden;
  }

  .c-iSBxEZ {
    vertical-align: middle;
    max-width: 100%;
  }

  .c-jxOsbj {
    height: 100%;
    width: 100%;
    object-fit: cover;
  }

  .c-dyvMgW {
    font-family: var(--fonts-body);
    font-weight: 400;
    margin: 0;
  }

  .c-dyvMgW > .c-dyvMgW:before,
  .c-dyvMgW > .c-dyvMgW:after {
    display: none;
  }

  .c-hMRxhp {
    display: inline-block;
    fill: none;
    flex-shrink: 0;
    stroke: currentcolor;
    stroke-linecap: round;
    stroke-linejoin: round;
    vertical-align: middle;
  }
}

@media  {
  .c-hMBZYr-kMhpXP-size-lg {
    height: var(--sizes-5);
    width: var(--sizes-5);
  }

  .c-dyvMgW-gvmVBy-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
  }

  .c-dyvMgW-gvmVBy-size-md::before {
    content: '';
    margin-bottom: -0.3864em;
    display: table;
  }

  .c-dyvMgW-gvmVBy-size-md::after {
    content: '';
    margin-top: -0.3864em;
    display: table;
  }

  .c-hMBZYr-fDXBFo-size-xs {
    height: var(--sizes-2);
    width: var(--sizes-2);
  }

  .c-hMRxhp-dMrjnF-size-md {
    height: var(--sizes-2);
    width: var(--sizes-2);
    stroke-width: 1.75;
  }

  .c-dyvMgW-bhKTXb-size-xs {
    font-size: var(--fontSizes-xs);
    line-height: 1.6;
  }

  .c-dyvMgW-bhKTXb-size-xs::before {
    content: '';
    margin-bottom: -0.4364em;
    display: table;
  }

  .c-dyvMgW-bhKTXb-size-xs::after {
    content: '';
    margin-top: -0.4364em;
    display: table;
  }
}

@media  {
  .c-dyvMgW-ildsiYK-css {
    color: var(--colors-tonal400);
  }

  .c-PJLV-ieGSWOZ-css {
    position: relative;
    height: 100%;
    width: 100%;
  }

  .c-hMRxhp-ijiyhIq-css {
    height: 100%;
    width: 100%;
  }
}

<div>
  <div
    class="c-hMBZYr c-hMBZYr-kMhpXP-size-lg"
  >
    <div
      class="c-PJLV c-PJLV-ieGSWOZ-css"
    >
      <svg
        aria-hidden="true"
        class="c-hMRxhp c-hMRxhp-dMrjnF-size-md c-hMRxhp-ijiyhIq-css"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.5 19.5c2.333-1 3.833-1.833 4.5-2.5 1-1-2-1-2-6 0-3.333 1.333-5 4-5s4 1.667 4 5c0 5-3 5-2 6 .667.667 2.167 1.5 4.5 2.5"
          stroke-linecap="round"
        />
        <circle
          cx="12"
          cy="12"
          r="10"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Avatar component renders an icon avatar 1`] = `
@media  {
  .c-hMBZYr {
    display: flex;
    justify-content: center;
    align-items: center;
    color: var(--colors-tonal400);
    border-radius: var(--radii-round);
    border: 2px solid var(--colors-tonal100);
    background-color: var(--colors-white);
    overflow: hidden;
  }

  .c-iSBxEZ {
    vertical-align: middle;
    max-width: 100%;
  }

  .c-jxOsbj {
    height: 100%;
    width: 100%;
    object-fit: cover;
  }

  .c-dyvMgW {
    font-family: var(--fonts-body);
    font-weight: 400;
    margin: 0;
  }

  .c-dyvMgW > .c-dyvMgW:before,
  .c-dyvMgW > .c-dyvMgW:after {
    display: none;
  }

  .c-hMRxhp {
    display: inline-block;
    fill: none;
    flex-shrink: 0;
    stroke: currentcolor;
    stroke-linecap: round;
    stroke-linejoin: round;
    vertical-align: middle;
  }
}

@media  {
  .c-hMBZYr-kMhpXP-size-lg {
    height: var(--sizes-5);
    width: var(--sizes-5);
  }

  .c-dyvMgW-gvmVBy-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
  }

  .c-dyvMgW-gvmVBy-size-md::before {
    content: '';
    margin-bottom: -0.3864em;
    display: table;
  }

  .c-dyvMgW-gvmVBy-size-md::after {
    content: '';
    margin-top: -0.3864em;
    display: table;
  }

  .c-hMBZYr-fDXBFo-size-xs {
    height: var(--sizes-2);
    width: var(--sizes-2);
  }

  .c-hMRxhp-dMrjnF-size-md {
    height: var(--sizes-2);
    width: var(--sizes-2);
    stroke-width: 1.75;
  }

  .c-dyvMgW-bhKTXb-size-xs {
    font-size: var(--fontSizes-xs);
    line-height: 1.6;
  }

  .c-dyvMgW-bhKTXb-size-xs::before {
    content: '';
    margin-bottom: -0.4364em;
    display: table;
  }

  .c-dyvMgW-bhKTXb-size-xs::after {
    content: '';
    margin-top: -0.4364em;
    display: table;
  }
}

@media  {
  .c-dyvMgW-ildsiYK-css {
    color: var(--colors-tonal400);
  }

  .c-PJLV-ieGSWOZ-css {
    position: relative;
    height: 100%;
    width: 100%;
  }

  .c-hMRxhp-ijiyhIq-css {
    height: 100%;
    width: 100%;
  }
}

<div>
  <div
    class="c-hMBZYr c-hMBZYr-kMhpXP-size-lg"
  >
    <svg
      aria-hidden="true"
      class="c-hMRxhp c-hMRxhp-dMrjnF-size-md"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 14v-4m-1-4H4v12h16zm-8 8v-4m-4 4v-4"
      />
    </svg>
  </div>
</div>
`;

exports[`Avatar component renders an image avatar 1`] = `
@media  {
  .c-hMBZYr {
    display: flex;
    justify-content: center;
    align-items: center;
    color: var(--colors-tonal400);
    border-radius: var(--radii-round);
    border: 2px solid var(--colors-tonal100);
    background-color: var(--colors-white);
    overflow: hidden;
  }

  .c-iSBxEZ {
    vertical-align: middle;
    max-width: 100%;
  }

  .c-jxOsbj {
    height: 100%;
    width: 100%;
    object-fit: cover;
  }
}

@media  {
  .c-hMBZYr-kMhpXP-size-lg {
    height: var(--sizes-5);
    width: var(--sizes-5);
  }
}

<div>
  <div
    class="c-hMBZYr c-hMBZYr-kMhpXP-size-lg"
  >
    <img
      alt="somobody's avatar"
      class="c-iSBxEZ c-jxOsbj"
      src="path-to-image"
    />
  </div>
</div>
`;

exports[`Avatar component renders an inital avatar 1`] = `
@media  {
  .c-hMBZYr {
    display: flex;
    justify-content: center;
    align-items: center;
    color: var(--colors-tonal400);
    border-radius: var(--radii-round);
    border: 2px solid var(--colors-tonal100);
    background-color: var(--colors-white);
    overflow: hidden;
  }

  .c-iSBxEZ {
    vertical-align: middle;
    max-width: 100%;
  }

  .c-jxOsbj {
    height: 100%;
    width: 100%;
    object-fit: cover;
  }

  .c-dyvMgW {
    font-family: var(--fonts-body);
    font-weight: 400;
    margin: 0;
  }

  .c-dyvMgW > .c-dyvMgW:before,
  .c-dyvMgW > .c-dyvMgW:after {
    display: none;
  }

  .c-hMRxhp {
    display: inline-block;
    fill: none;
    flex-shrink: 0;
    stroke: currentcolor;
    stroke-linecap: round;
    stroke-linejoin: round;
    vertical-align: middle;
  }
}

@media  {
  .c-hMBZYr-kMhpXP-size-lg {
    height: var(--sizes-5);
    width: var(--sizes-5);
  }

  .c-dyvMgW-gvmVBy-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
  }

  .c-dyvMgW-gvmVBy-size-md::before {
    content: '';
    margin-bottom: -0.3864em;
    display: table;
  }

  .c-dyvMgW-gvmVBy-size-md::after {
    content: '';
    margin-top: -0.3864em;
    display: table;
  }

  .c-hMBZYr-fDXBFo-size-xs {
    height: var(--sizes-2);
    width: var(--sizes-2);
  }

  .c-hMRxhp-dMrjnF-size-md {
    height: var(--sizes-2);
    width: var(--sizes-2);
    stroke-width: 1.75;
  }

  .c-dyvMgW-bhKTXb-size-xs {
    font-size: var(--fontSizes-xs);
    line-height: 1.6;
  }

  .c-dyvMgW-bhKTXb-size-xs::before {
    content: '';
    margin-bottom: -0.4364em;
    display: table;
  }

  .c-dyvMgW-bhKTXb-size-xs::after {
    content: '';
    margin-top: -0.4364em;
    display: table;
  }
}

@media  {
  .c-dyvMgW-ildsiYK-css {
    color: var(--colors-tonal400);
  }

  .c-PJLV-ieGSWOZ-css {
    position: relative;
    height: 100%;
    width: 100%;
  }

  .c-hMRxhp-ijiyhIq-css {
    height: 100%;
    width: 100%;
  }
}

<div>
  <div
    class="c-hMBZYr c-hMBZYr-fDXBFo-size-xs"
  >
    <p
      class="c-dyvMgW c-dyvMgW-bhKTXb-size-xs c-dyvMgW-ildsiYK-css"
    >
      A
    </p>
  </div>
</div>
`;

exports[`Avatar component renders an xs size with the fallback icon when src and name are missing 1`] = `
@media  {
  .c-hMBZYr {
    display: flex;
    justify-content: center;
    align-items: center;
    color: var(--colors-tonal400);
    border-radius: var(--radii-round);
    border: 2px solid var(--colors-tonal100);
    background-color: var(--colors-white);
    overflow: hidden;
  }

  .c-iSBxEZ {
    vertical-align: middle;
    max-width: 100%;
  }

  .c-jxOsbj {
    height: 100%;
    width: 100%;
    object-fit: cover;
  }

  .c-dyvMgW {
    font-family: var(--fonts-body);
    font-weight: 400;
    margin: 0;
  }

  .c-dyvMgW > .c-dyvMgW:before,
  .c-dyvMgW > .c-dyvMgW:after {
    display: none;
  }

  .c-hMRxhp {
    display: inline-block;
    fill: none;
    flex-shrink: 0;
    stroke: currentcolor;
    stroke-linecap: round;
    stroke-linejoin: round;
    vertical-align: middle;
  }
}

@media  {
  .c-hMBZYr-kMhpXP-size-lg {
    height: var(--sizes-5);
    width: var(--sizes-5);
  }

  .c-dyvMgW-gvmVBy-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
  }

  .c-dyvMgW-gvmVBy-size-md::before {
    content: '';
    margin-bottom: -0.3864em;
    display: table;
  }

  .c-dyvMgW-gvmVBy-size-md::after {
    content: '';
    margin-top: -0.3864em;
    display: table;
  }

  .c-hMBZYr-fDXBFo-size-xs {
    height: var(--sizes-2);
    width: var(--sizes-2);
  }

  .c-hMRxhp-dMrjnF-size-md {
    height: var(--sizes-2);
    width: var(--sizes-2);
    stroke-width: 1.75;
  }
}

@media  {
  .c-dyvMgW-ildsiYK-css {
    color: var(--colors-tonal400);
  }

  .c-PJLV-ieGSWOZ-css {
    position: relative;
    height: 100%;
    width: 100%;
  }

  .c-hMRxhp-ijiyhIq-css {
    height: 100%;
    width: 100%;
  }
}

<div>
  <div
    class="c-hMBZYr c-hMBZYr-fDXBFo-size-xs"
  >
    <div
      class="c-PJLV c-PJLV-ieGSWOZ-css"
    >
      <svg
        aria-hidden="true"
        class="c-hMRxhp c-hMRxhp-dMrjnF-size-md c-hMRxhp-ijiyhIq-css"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.5 19.5c2.333-1 3.833-1.833 4.5-2.5 1-1-2-1-2-6 0-3.333 1.333-5 4-5s4 1.667 4 5c0 5-3 5-2 6 .667.667 2.167 1.5 4.5 2.5"
          stroke-linecap="round"
        />
        <circle
          cx="12"
          cy="12"
          r="10"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Avatar component renders the initial of the name when the src of the image is not available and a name was provided 1`] = `
@media  {
  .c-hMBZYr {
    display: flex;
    justify-content: center;
    align-items: center;
    color: var(--colors-tonal400);
    border-radius: var(--radii-round);
    border: 2px solid var(--colors-tonal100);
    background-color: var(--colors-white);
    overflow: hidden;
  }

  .c-iSBxEZ {
    vertical-align: middle;
    max-width: 100%;
  }

  .c-jxOsbj {
    height: 100%;
    width: 100%;
    object-fit: cover;
  }

  .c-dyvMgW {
    font-family: var(--fonts-body);
    font-weight: 400;
    margin: 0;
  }

  .c-dyvMgW > .c-dyvMgW:before,
  .c-dyvMgW > .c-dyvMgW:after {
    display: none;
  }
}

@media  {
  .c-hMBZYr-kMhpXP-size-lg {
    height: var(--sizes-5);
    width: var(--sizes-5);
  }

  .c-dyvMgW-gvmVBy-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
  }

  .c-dyvMgW-gvmVBy-size-md::before {
    content: '';
    margin-bottom: -0.3864em;
    display: table;
  }

  .c-dyvMgW-gvmVBy-size-md::after {
    content: '';
    margin-top: -0.3864em;
    display: table;
  }
}

@media  {
  .c-dyvMgW-ildsiYK-css {
    color: var(--colors-tonal400);
  }
}

<div>
  <div
    class="c-hMBZYr c-hMBZYr-kMhpXP-size-lg"
  >
    <p
      class="c-dyvMgW c-dyvMgW-gvmVBy-size-md c-dyvMgW-ildsiYK-css"
    >
      B
    </p>
  </div>
</div>
`;
